ios/FluentUI/Navigation/NavigationBar.swift

```diff
@@ -398,12 +462,29 @@ open class NavigationBar: UINavigationBar {
         }
     }
 
+    @objc private func updateNavigationItem(notification: NSNotification) {
+        let item = notification.object as? UINavigationItem
+        if let item = item {
+            update(with: item)
+        }
+    }
+
     func update(with navigationItem: UINavigationItem) {
         let (actualStyle, actualItem) = actualStyleAndItem(for: navigationItem)
         style = actualStyle
         updateColors(for: actualItem)
         showsLargeTitle = navigationItem.usesLargeTitle
         updateShadow(for: navigationItem)
+        updateTopAccessoryView(for: navigationItem)
+
+        NotificationCenter.default.removeObserver(self,
+                                                  name: UINavigationItem.navigationItemDidChangeNotification,
+                                                  object: nil)
+
+        NotificationCenter.default.addObserver(self,
```

Comment:
I am bit confusing. what are we lisenting on top of leftBarButtonItemObserver, rightbarButtonItemsObserver etc?

Reply:
Notifications or observers end up doing the same thing here.
Because we are using property observers for other things, I update the logic to use those instead of notifications.
I tested that each property updates correctly after being set to new values after the nav bar is presented.